spec/services/publish_draft_edition_service_spec.rb

```diff
@@ -31,15 +37,23 @@
     end
 
     context "when there is a live edition" do
+      let(:document) { create(:document, :with_current_and_live_editions) }
+      let(:current_edition) { document.current_edition }
+
       it "supersedes the live edition" do
-        document = create(:document, :with_current_and_live_editions)
-        current_edition = document.current_edition
         live_edition = document.live_edition
 
         described_class.call(current_edition, user, with_review: true)
         expect(document.live_edition).to eq(current_edition)
         expect(live_edition).to be_superseded
       end
+
+      it "doesn't overwrite when the document was first published" do
+        first_published_at = document.reload.first_published_at
+        described_class.call(current_edition, user, with_review: true)
+
+        expect(document.first_published_at).to eq(first_published_at)
```

Comment:
I think you can get around the reload by doing 

```
        expect { described_class.call(current_edition, user, with_review: true) }
          .not_to change(document, :first_published_at)
```

Reply:
Good call, that is a lot nicer!